mantidproject/mantid
# Save Instrument from memory to disk for NexusGeometry

Fixes #25942 

For ongoing implementation of added feature to write out instrument in NexusGeometryParser.

'NexusGeometrySave.cpp' **=>** Void function saveInstrument taking (Geometry::ComponentInfo) component information and (std::string) full path as arguments.

'NexusGeometrySaveTest.h' **=>** destructive testing calling saveInstrument with false parameters.

**Report to:** 
-->OwenArnold

**Tester**
* Detailed Code Review
* Check test coverage and quality
* Verify that we can go between `saveInstrument` and `NexusGeometryParser` loading seamlessly. Note that we will not be saving out shape information as part of this PR.

Please review. Do not merge until passes CI

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
